mimic/rest/nova_api.py

```diff
@@ -399,6 +410,40 @@ def perform_action(self, request, tenant_id, server_id):
         """
         return self._region_collection_for_tenant(tenant_id).request_action(request, server_id, self.url)
 
+    @app.route("/v2/<string:tenant_id>/os-keypairs", methods=['GET'])
+    def get_key_pairs(self, request, tenant_id):
+        """
+        Returns current key pairs
+        """
+        return self._keypair_collection_for_tenant(tenant_id).json_list()
+
+    @app.route("/v2/<string:tenant_id>/os-keypairs", methods=['POST'])
+    def create_key_pair(self, request, tenant_id):
+        """
+        Returns a newly created key pair with the specified name.
+        """
+        try:
+            content = json.loads(request.content.read())
+        except ValueError:
+            request.setResponseCode(400)
+            return json.dumps(bad_request("Malformed request body", request))
+
+        keypair = content["keypair"]
+        keypair_from_request = KeyPair(
+            name=keypair["name"], public_key=keypair["public_key"])
```

Comment:
This will cause an exception if the `name` or `public_key` keys do not exist. Can we please handle this.